utils/iterator/empty.go

```diff
@@ -0,0 +1,23 @@
+// Copyright (C) 2019-2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package iterator
+
+import "github.com/ava-labs/avalanchego/utils"
+
+// Empty returns an iterator with no elements.
+func Empty[T any]() Iterator[T] {
+	return &empty[T]{}
+}
+
+type empty[T any] struct{}
```

Comment:
Would it make more sense to just have:
```suggestion
// Empty is an iterator with no elements.
type Empty[T any] struct{}
```
Doesn't seem like the `New` function buys us anything